File: Android Developer Fundamentals Course (V1)/HelloContraint/app/src/main/java/com/example/luispena/helloconstraint/MainActivity.java
package com.example.luispena.helloconstraint;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    private int mCount = 0;
    private TextView mShowCount;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


        //set
        mShowCount = (TextView) findViewById(R.id.show_count);

    }

    public void showToast(View view) {

        Toast toast = Toast.makeText(this, R.string.button_label_toast, Toast.LENGTH_SHORT);
        toast.show();

    }

    public void countUp(View view) {

        mCount++;
        if (mShowCount != null) {

            mShowCount.setText(Integer.toString(mCount));

        }


    }
}

File: KotlinCodeLabs/DiceRoller - 01.2/app/src/main/java/com/example/diceroller/MainActivity.kt
package com.example.diceroller

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*
import java.util.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {

        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        roll_button.setOnClickListener{ roll()}
        count_button.setOnClickListener{ roll()}



    }

    private fun roll(){
        val resultText: TextView = findViewById(R.id.result_text)
        val randomInt = Random().nextInt(6) + 1
        Toast.makeText(this,"Button Clicked", Toast.LENGTH_SHORT).show()
        resultText.text = randomInt.toString()

        count(randomInt)
    }

    private fun count(arg: Int){

        val countButton : Button = findViewById(R.id.count_button)
        val resultCount: TextView = findViewById(R.id.result_count)

        if (arg == 6){

            val countResult: Int = arg
            resultCount.text = countResult.toString()


        }else {
            val countResult: Int = arg + 1
            resultCount.text = countResult.toString()


        }

        Toast.makeText(this,"Button Clicked", Toast.LENGTH_SHORT).show()

    }



}
